src/Bicep.Core/Semantics/NameBindingVisitor.cs

```diff
@@ -44,17 +48,17 @@ public override void VisitProgramSyntax(ProgramSyntax syntax)
             // create bindings for all of the declarations to their corresponding symbol
             // this is needed to make find all references work correctly
             // (doing this here to avoid side-effects in the constructor)
-            foreach (DeclaredSymbol declaredSymbol in this.declarations.Values)
+            foreach (DeclaredSymbol declaredSymbol in this.outermostDeclarations.Values)
             {
                 this.bindings.Add(declaredSymbol.DeclaringSyntax, declaredSymbol);
             }
 
             // include all the locals in the symbol table as well
             // since we only allow lookups by object and not by name,
             // a flat symbol table should be sufficient
-            foreach (var declaredSymbol in allLocalScopes.Values.SelectMany(scope => scope.AllDeclarations))
+            foreach (var declaredSymbol in allLocalScopes.Values.SelectMany(scope => scope.Declarations))
             {
-                this.bindings.Add(declaredSymbol.DeclaringSyntax, declaredSymbol);
+                this.bindings.TryAdd(declaredSymbol.DeclaringSyntax, declaredSymbol);

```

Comment:
>TryAdd [](start = 30, length = 6)

Why TryAdd? I wanted Add to throw if somehow we're inserting duplicates.

Reply:
I think this probably can be reverted. This part of the change might be left over from one of my other ways of factoring symbols/scopes.